### PR TITLE
chore: add apache 2 license to repo

### PR DESCRIPTION
**Pull request by mmalina**
License was missing in this repo. Add Apache 2 license. This was brought up by mbasti, thanks!

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (davidmogar) [inline LICENSE:189]: 🙃

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Comment (davidmogar) [inline LICENSE:178]: 🙃

**Review by davidmogar — CHANGES_REQUESTED** [review]

File: LICENSE (hunk 1 above)
Comment (mmalina) [reply LICENSE]: I didn't realize you need to change something in the template. But FWIW, we have the same issue here :) https://github.com/redhat-appstudio/release-service-catalog/blob/main/LICENSE and here https://github.com/redhat-appstudio/release-service/blob/main/LICENSE

Fixed here.

File: LICENSE (hunk 2 above)
Comment (mmalina) [reply LICENSE:178]: Hmm, what shall we do about this? I guess you're supposed to add that snippet to all your files ideally. But it's probably generally accepted to just have the license file in the repo and then it's implied that it's all covered by it. And here we also have that snippet left there: https://github.com/redhat-appstudio/infra-deployments/blob/main/LICENSE

So I'd say leave it there?

Comment (davidmogar) [reply LICENSE:178]: Can be omitted, but fair enough. Let's keep it.

**Review by davidmogar — APPROVED** [review]

**Pull request merged**